### PR TITLE
SaaS integration tests cleanup

### PR DESCRIPTION
**Pull request by galvana**
Closes [PROD-1877](https://ethyca.atlassian.net/browse/PROD-1877)

### Description Of Changes

Misc fixes for the SaaS integration tests. Added inline comments throughout explaining the changes

### Code Changes

* [ ] See PR comments

### Steps to Confirm

* [ ] The SaaS integration tests should pass

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* [ ] Update `CHANGELOG.md`

[PROD-1877]: https://ethyca.atlassian.net/browse/PROD-1877?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: data/saas/config/aircall_config.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ saas_config:
 
   test_request:
     method: GET
-    path: /v1/company/
+    path: /v1/company
```
Comment (galvana) [inline data/saas/config/aircall_config.yml:32]: Aircall doesn't like the trailing slash anymore 🤷‍♂️

File: pyproject.toml [hunk 2]
```diff
@@ -185,32 +185,9 @@ markers = [
...
-    "integration_braintree",
-    "integration_auth0",
-    "integration_shopify",
-    "integration_square",
-    "integration_outreach",
-    "integration_recharge",
-    "integration_salesforce",
-    "integration_twilio_conversations",
-    "integration_adobe_campaign",
-    "integration_firebase_auth",
```
Comment (galvana) [inline pyproject.toml:212]: Removing these old pytest markers for individual tests. There wasn't a need to continue doing this so I'm just cleaning them up.

File: tests/fixtures/saas/amplitude_fixtures.py [hunk 3]
```diff
@@ -170,5 +170,7 @@ def amplitude_create_erasure_data(
     amplitude_test_client.create_user(amplitude_erasure_identity_email)
 
     poll_for_existence(
-        amplitude_test_client.get_user, (amplitude_erasure_identity_email,)
+        amplitude_test_client.get_user,
+        (amplitude_erasure_identity_email,),
+        verification_count=3,
```
Comment (galvana) [inline tests/fixtures/saas/amplitude_fixtures.py:175]: Changing this fixture to wait for 3 successful responses before considering the test data "seeded"

File: tests/fixtures/saas/auth0_fixtures.py [hunk 4]
```diff
@@ -129,7 +130,7 @@ def auth0_access_data(
 
     headers = {"Authorization": f"Bearer {auth0_token}"}
     user_response = requests.get(
-        url=f"{base_url}/api/v2/users-by-email?email={auth0_identity_email}",
+        url=f"{base_url}/api/v2/users-by-email?email={quote(auth0_identity_email)}",
```
Comment (galvana) [inline tests/fixtures/saas/auth0_fixtures.py:133]: Escaping the `+` in the `auth0_identity_email`

File: tests/fixtures/saas/mailchimp_override_fixtures.py [hunk 5]
```diff
@@ -21,16 +21,14 @@
...
-    )
+    return load_config("tests/fixtures/saas/test_data/mailchimp_override_config.yml")
 
 
 @pytest.fixture
 def mailchimp_override_dataset() -> Dict[str, Any]:
-    return load_dataset(
-        "data/saas/dataset/request_override/mailchimp_override_dataset.yml"
-    )[0]
+    return load_dataset("tests/fixtures/saas/test_data/mailchimp_override_dataset.yml")[
```
Comment (galvana) [inline tests/fixtures/saas/mailchimp_override_fixtures.py:29]: Path change

File: tests/fixtures/saas/stripe_fixtures.py [hunk 6]
```diff
@@ -292,7 +291,7 @@ def stripe_create_erasure_data(
                 "card": {
                     "number": 4242424242424242,
                     "exp_month": 4,
-                    "exp_year": 2023,
+                    "exp_year": datetime.today().year + 1,
```
Comment (galvana) [inline tests/fixtures/saas/stripe_fixtures.py:294]: 2023 seemed so far in the future...

File: tests/fixtures/saas/stripe_fixtures.py [hunk 7]
```diff
@@ -153,7 +153,6 @@ def stripe_create_erasure_data(
                 "state": "CA",
             },
             "name": "Ethyca RTF",
-            "phone": "+19515551234",
```
Comment (galvana) [inline tests/fixtures/saas/stripe_fixtures.py:156]: This test number is used for the access customer so I'm removing it from the auto-generated erasure customer

File: tests/fixtures/saas_erasure_order_fixtures.py [hunk 8]
```diff
@@ -33,7 +33,7 @@ def saas_erasure_order_config() -> Dict:
 @pytest.fixture
 def saas_erasure_order_dataset() -> Dict:
     return load_dataset_with_replacement(
-        "data/saas/dataset/saas_erasure_order_dataset.yml",
+        "tests/fixtures/saas/test_data/saas_erasure_order_dataset.yml",
```
Comment (galvana) [inline tests/fixtures/saas_erasure_order_fixtures.py:36]: Another path change

File: tests/fixtures/saas_example_fixtures.py [hunk 9]
```diff
@@ -15,6 +15,7 @@
 from fides.api.models.datasetconfig import DatasetConfig
 from fides.api.models.policy import Policy, Rule, RuleTarget
 from fides.api.models.sql_models import Dataset as CtlDataset
+from fides.api.schemas.policy import ActionType
```
Comment (galvana) [inline tests/fixtures/saas_example_fixtures.py:18]: Adding missing import

File: tests/ops/api/v1/endpoints/test_connection_config_endpoints.py [hunk 10]
```diff
@@ -925,7 +925,6 @@ def test_filter_test_status(
         assert [item["last_test_succeeded"] is None for item in items]
 
     @pytest.mark.integration_saas
-    @pytest.mark.integration_stripe
```
Comment (galvana) [inline tests/ops/api/v1/endpoints/test_connection_config_endpoints.py:928]: A lot of these changes are just the removal of the individual pytest marks

File: tests/ops/integration_tests/limiter/test_rate_limiter.py
Comment (galvana) [inline tests/ops/integration_tests/limiter/test_rate_limiter.py:1]: We don't have access to Zendesk anymore, so I swapped it out with something we still have access to, Stripe

File: tests/ops/integration_tests/saas/test_friendbuy_task.py [hunk 11]
```diff
@@ -17,16 +17,16 @@
 logger = logging.getLogger(__name__)
 
 
+@pytest.mark.skip(reason="No active account")
```
Comment (galvana) [inline tests/ops/integration_tests/saas/test_friendbuy_task.py:20]: The other common change, just disabling tests we don't have active accounts for

File: tests/ops/integration_tests/test_connection_configuration_integration.py [hunk 12]
```diff
@@ -1200,20 +1199,17 @@ def test_saas_connection_connect_with_extra_secrets(
...
         resp = api_client.put(
             url,
             headers=auth_header,
             json=payload,
         )
-        assert resp.status_code == 422
-
-        body = json.loads(resp.text)
-        assert body["detail"][0]["msg"] == "extra fields not permitted"
+        assert resp.status_code == 200
```
Comment (galvana) [inline tests/ops/integration_tests/test_connection_configuration_integration.py:1209]: Missed this once since it was hidden behind the "unsafe tests" flag

**Review by Kelsey-Ethyca — APPROVED** [review]
🦖

**Pull request merged**